cpp/src/arrow/dataset/scanner_test.cc

```diff
@@ -706,30 +634,6 @@ TEST_P(TestScanner, Head) {
   AssertTablesEqual(*expected, *actual, /*same_chunk_layout=*/false);
 }
 
-TEST_P(TestScanner, FromReader) {
```

Comment:
Should this test be kept? The API still exists/is still used and it seems it's not true that the async scanner can't be constructed from a reader.

Reply:
Good catch.  This test ended up working just fine on an async scanner.